### PR TITLE
Mm 44766 participants

### PR DESCRIPTION
**Pull request by gbochora**
#### Summary
This PR adds `RHS > Participants` view:
- Participants list
- `RHS` back navigation
- `Send message` button on hover 

NOTE: Search is not implemented yet. We decided to do it in v2.

https://user-images.githubusercontent.com/4368372/176377807-fddbdfe7-9cb6-442b-97eb-87493094bbe5.mp4

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44766

#### Checklist

~~- [ ] Telemetry updated~~
~~- [ ] Gated by experimental feature flag~~
- [ ] Unit tests updated

### CODE REVIEW

File: webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx [hunk 1]
```diff
@@ -75,13 +75,13 @@ export const RunHeader = ({playbookRun, playbookRunMetadata, openRHS, role}: Pro
                 tooltipId={'timeline-button-tooltip'}
                 tooltipMessage={formatMessage({defaultMessage: 'View Timeline'})}
                 className={'icon-update'}
-                onClick={() => openRHS(RHSContent.RunTimeline, formatMessage({defaultMessage: 'Timeline'}), playbookRun.name)}
+                onClick={() => onViewTimeline()}
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx:78]:
```suggestion
                onClick={onViewTimeline}
```

File: webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx [hunk 2]
```diff
@@ -75,13 +75,13 @@ export const RunHeader = ({playbookRun, playbookRunMetadata, openRHS, role}: Pro
...
             />
             <HeaderButton
                 tooltipId={'info-button-tooltip'}
                 tooltipMessage={formatMessage({defaultMessage: 'View Info'})}
                 className={'icon-information-outline'}
-                onClick={() => openRHS(RHSContent.RunInfo, formatMessage({defaultMessage: 'Run info'}), playbookRun.name)}
+                onClick={() => onViewInfo()}
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx:84]:
```suggestion
                onClick={onViewInfo}
```

File: webapp/src/components/backstage/playbook_runs/playbook_run/rhs.tsx [hunk 3]
```diff
@@ -40,6 +41,14 @@ const RightHandSidebar = ({isOpen, onClose, title, children, subtitle}: Props) =
             isOpen={isOpen}
         >
             <Header>
+                {
+                    onBack ? <BackIcon>
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/rhs.tsx:45]: this is a personal preference, but you could use a better indentation when doing ternary operator.

Example.

```ts 
            {props.parent === 'retro' ? (
                <TimeContainer parent={props.parent}>
                    {timeSincePrevEvent}
                </TimeContainer>
            ) : null}
```

**Review by trilopin — CHANGES_REQUESTED** [review]
Thank you @gbochora 

I left some details and doubts

File: webapp/src/components/backstage/playbook_runs/playbook_run/send_message_button.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    teamName: string | null;
+}
+
+export const SendMessageButton = (props: Props) => {
+    const dispatch = useDispatch();
+    const user = useSelector<GlobalState, UserProfile>((state) => getUser(state, props.userId));
+
+    useEffect(() => {
+        if (!user) {
+            dispatch(fetchUser(props.userId));
```
[the remark this answers is not in the document]
Comment (gbochora) [reply webapp/src/components/backstage/playbook_runs/playbook_run/send_message_button.tsx]: Good point. I moved fetching inside the `onClick` function.

File: webapp/src/components/backstage/playbook_runs/playbook_run/send_message_button.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        if (user) {
+            navigateToUrl(`/${props.teamName}/messages/@${user.username}`);
+            return;
+        }
+
+        // if user profile is missing fetch it
+        const actionFunc = dispatch(fetchUser(props.userId));
+
+        //@ts-ignore
+        actionFunc.then((data) => {
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/send_message_button.tsx:36]: I tested this and works fine. But as far as I know, waiting for a dispatched action does not guarantee that the reducer has been executed. 

what happens if you navigate without doing the fetch first?

**Review by trilopin — COMMENTED** [review]
I left a comment about promises and actions. Additionally, I'm wondering if we should avoid navigating to our own user.

File: webapp/src/components/backstage/playbook_runs/playbook_run/send_message_button.tsx (hunk 5 above)
Comment (trilopin) [reply webapp/src/components/backstage/playbook_runs/playbook_run/send_message_button.tsx:36]: not the selector but the action.

I think with redux, the fact that the promise of the action has been solved doesn't mean that the reducer has been executed. 

I'm still not sure if we expected that or if we simply do a best-effort strategy (that would be enough most of the cases).

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Impressive work! The only blocking comment is the one about the plural, because if there's one participant, it still reads "1 participants". All the other comments are nits. Thanks! :tada:

File: webapp/src/components/backstage/playbook_runs/playbook_run/rhs_participants.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        <Container>
+            <SearchSection>
+                {/* <StyledSearchInput
+                    testId={'search-filter'}
+                    default={''}
+                    onSearch={setSearchTerm}
+                    placeholder={formatMessage({defaultMessage: 'Search'})}
+                /> */}
+                <ParticipantsNumber>
+                    {formatMessage({defaultMessage: '{num} Participants'}, {num: participantsIds.length})}
```
[the remark this answers is not in the document]
Comment (gbochora) [reply webapp/src/components/backstage/playbook_runs/playbook_run/rhs_participants.tsx:36]: Nice catch!

**Review by trilopin — APPROVED** [review]
🎉

**Pull request merged**